spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/config/DiscoveryClientOptionalArgsConfiguration.java

```diff
@@ -56,6 +58,12 @@ public TlsProperties tlsProperties() {
 		return new TlsProperties();
 	}
 
+	@Bean
```

Comment:
I'm thinking this is not necessary. Since, in general, we would expect the users to make changes in a properties file, rather than creating their own beans, I'm thinking you could change `RestTemplateTimeoutProperties` from interface to class (remove `DefaultRestTemplateTimeoutProperties`) and move its content to `RestTemplateTimeoutProperties`, and annotate that class with `@ConfigurationProperties`, with the appropriate prefix.